pkgs/by-name/ic/icann-rdap/package.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  perl,
+  writableTmpDirAsHomeHook,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "icann-rdap";
+  version = "0.0.22";
+
+  src = fetchFromGitHub {
+    owner = "icann";
+    repo = "icann-rdap";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-J8uvFTjFY7YrohxlD7UwzrQUUoHHEb5mzZoQ64XUQHY=";
+  };
+
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-2R6GrcuksEOk8GiVkFhMljS12V2n0J1rCw9MCOtwMjA=";
+
+  nativeBuildInputs = [ perl ];
```

```suggestion
  nativeBuildInputs = [ pkg-config ];

  buildInputs = [ openssl ];
  
  env.OPENSSL_NO_VENDOR = true;
```

Don't vendor openssl